src/app/content/search/reducer.ts

```diff
@@ -34,11 +33,6 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
     case getType(actions.clearSearch): {
       return initialState;
     }
-    case getType(locationChange): {
-      return action.payload.action === 'PUSH' && !action.payload.location.state.search
```

Comment:
why would we not want to clear the search data when navigating with a PUSH and not having a new search? you'd have to change this to look in the query instead of the state, but it seems probably still valid

Reply:
When you click on a figure link then there will be PUSH action dispatched with no query and just hash
This would mean that we will remove the search results which doesn't seem good
But maybe we should clear this state when `book` id does not match with selected results id?
```ts
case getType(locationChange): {
      return state.selectedResult
        && action.payload.action === 'PUSH'
        && action.payload.location.state.pageUid !== stripIdVersion(state.selectedResult.result.source.pageId)
          ? initialState
          : state;
    }
```

Reply:
Actually, this is clearing results when clicking on a search result that is on another page so this is also not good